src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/ProvisionedInfrastructure.kt

```diff
@@ -0,0 +1,34 @@
+package com.atlassian.performance.tools.awsinfrastructure.api
+
+import com.atlassian.performance.tools.aws.api.Resource
+import com.atlassian.performance.tools.aws.api.UnallocatedResource
+import com.atlassian.performance.tools.infrastructure.api.virtualusers.VirtualUsers
+
+class ProvisionedInfrastructure<out T : VirtualUsers>
+@Deprecated("Use ProvisionedInfrastructure.Builder instead.")
+constructor(
+    val infrastructure: Infrastructure<T>,
+    val resource: Resource
+) {
+    object Defaults {
```

Comment:
What's the upside over the inlined defaults?

Reply:
You can't wrap (e.g. monitor) inlined defaults

Reply:
Good use case, I needed it a few times in the past :+1: